fiasco/tests/test_ion.py

```diff
@@ -158,7 +158,7 @@ def test_proton_collision(fe10):
 def test_missing_abundance(hdf5_dbase_root):
     ion = fiasco.Ion('Li 1',
                      temperature,
-                     abundance_filename='sun_coronal_1992_feldman',
+                     abundance='sun_coronal_1992_feldman',
```

Comment:
I'll give that a shot.  I think you could also just move the `with pytest.raises(KeyError):` to the top of this function and remove the `_ = ion.abundance` line.

Reply:
I just tried moving the logic to the setter, which works fine, but the abundance might still need to be instantiated in the constructor.  I get a funny little bug otherwise:
```
>>> iron = fiasco.Ion('Fe I', temperature = 1e6*u.K, abundance='sun_coronal_1992_feldman')
>>> iron.abundance
'sun_coronal_1992_feldman'
```

Reply:
Yep that's right. You'll still need a line in `__init__` that sets the abundance to some value so that the setter is triggered.

Reply:
Because of this, I've changed the function as I suggested before.  It will fail the test for the same reason as my initial commit otherwise.

Reply:
Ah, yes you're right. My initial comment was wrong since you of course want to trigger the setter in the constructor.